units/host_setup_script.go

```diff
@@ -127,3 +130,12 @@ func (j *hostSetupScriptJob) tryRequeue(ctx context.Context) error {
 	return j.env.RemoteQueue().Put(ctx, NewHostSetupScriptJob(j.env, j.host, j.CurrentAttempt+1))
 
 }
+
+func runSpawnHostSetupScript(ctx context.Context, env evergreen.Environment, h *host.Host) error {
+	script := fmt.Sprintf("cd %s\n%s", h.Distro.HomeDir(), h.ProvisionOptions.SetupScript)
```

Comment:
Could a parameter of the job set jasper's working dir instead of cd'ing?
(and ssh puts us there anyway)

Reply:
It'll add some complexity and I don't think it's very possible - if we're using legacy SSH, we'll still have to resort to `cd ${my_script_working_directory}` at the beginning of the script. This is further complicated by Windows, because it has to use a native path for the working directory (but not if it's legacy SSH on Windows). That means that all calls to `NewHostSetupScript` will have to if-else switch on whether or not the host is using legacy SSH or not.

Reply:
I don't understand, though: if NewHostExecuteJob takes a parameter for working dir, couldn't NewHostSetupScript jobs just pass `h.Distro.HomeDir()`? The logic to switch to that directory could be contained with the NewHostExecuteJob.

Also, in legacy SSH, doesn't ssh put us in the user's home directory without us doing anything (and then bash starts there as well)

Reply:
It wouldn't really be bulletproof because the working directory is not guaranteed to be $HOME, because the host execute job is allowed to run on any distro for any purpose.

1. For Windows, what happens if you specify a path outside of Cygwin (e.g. I want to start in `Z:/`, do I specify it as the cygwin `/cygdrive/z` or native `Z:/`)?
2. Legacy SSH starts in $HOME but you would still have to cd to the correct place if it's not $HOME.